packages/core/snapshot/write.go

```diff
@@ -109,6 +122,20 @@ func streamSnapshotDataTo(
 	return header, nil
 }
 
+// NewSolidEntryPointsProducer returns a SolidEntryPointsProducerFunc that provide solid entry points from the snapshot manager.
+func NewSolidEntryPointsProducer(lastConfirmedEpoch, latestCommitableEpoch epoch.Index, smgr *Manager) SolidEntryPointsProducerFunc {
+	prodChan := make(chan *SolidEntryPoints)
+	smgr.SnapshotSolidEntryPoints(lastConfirmedEpoch, latestCommitableEpoch, prodChan)
+
+	return func() *SolidEntryPoints {
+		obj, ok := <-prodChan
```

Comment:
I think we have a problem with this Producer: you are checking if there is data in the channel, if not you are returning nil. But since you are writing to the channel from a goroutine, there is not guarantee that there is data in it at any point in time when you are reading it; in which case you will only obtain partial data!
I've checked and the other "producers" that you have wrote as part of the streameable snapshot PR suffer from the same issue!
Can we simplify these producers? Do they really need a channel after all? They can maybe just be synchronous normal function calls without a goroutine.

Reply:
Hmm... I see your point, however at the time we actually get data from `prodChan`, it should be filled with data already.